### PR TITLE
Fix conditions and task privileges

### PR DESCRIPTION
**Pull request by rabi**

### CODE REVIEW

File: pkg/deployment/openstack.go [hunk 1]
```diff
@@ -140,6 +140,7 @@ func RunOpenStack(ctx context.Context, helper *helper.Helper, obj client.Object,
 		Strategy:       "linear",
 		GatherFacts:    false,
 		AnyErrorsFatal: true,
+		Become:         true,
```
Comment (bshephar) [inline pkg/deployment/openstack.go:143]: Do you envision that we will do this temporarily until we have ironed out the privilege escalation stuff in edpm_ansible?

Comment (rabi) [reply pkg/deployment/openstack.go:143]: Yes and also these would be moving to dataplaneservice interface/CRs soon. This is a WA for the time being to unblock.

Comment (rabi) [reply pkg/deployment/openstack.go:143]: We could have removed https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/pkg/deployment/os.go#L203, but that breaks a few roles. So the plan is to fix roles not to require become:true at the play level.

Comment (rabi) [reply pkg/deployment/openstack.go:143]: https://github.com/openstack-k8s-operators/edpm-ansible/pull/158 would be probably enough to remove Become:true from RunOS tasks. We'll see.

Comment (rabi) [reply pkg/deployment/openstack.go:143]: Looks like more work required with some other roles which I'll work next week. We should merge this IMO.

**Comment by slagle** [comment]
The change LGTM, but can you fix the kuttl tests? This should do it:

```
diff --git a/tests/kuttl/tests/dataplane-create-test/00-assert.yaml b/tests/kuttl/tests/dataplane-create-test/00-assert.yaml
index c3c4882..66025b0 100644
--- a/tests/kuttl/tests/dataplane-create-test/00-assert.yaml
+++ b/tests/kuttl/tests/dataplane-create-test/00-assert.yaml
@@ -149,14 +149,6 @@ status:
     severity: Info
     status: "False"
     type: Ready
-  - message: Init
-    reason: Init
-    status: Unknown
-    type: ConfigureCephClientReady
-  - message: Init
-    reason: Init
-    status: Unknown
-    type: ConfigureNetworkReady
   - message: Init
     reason: Init
     status: Unknown
@@ -235,14 +227,6 @@ status:
     severity: Info
     status: "False"
     type: Ready
-  - message: Init
-    reason: Init
-    status: Unknown
-    type: ConfigureCephClientReady
-  - message: Init
-    reason: Init
-    status: Unknown
-    type: ConfigureNetworkReady
   - message: Init
     reason: Init
     status: Unknown
@@ -317,14 +301,6 @@ status:
     severity: Info
     status: "False"
     type: Ready
-  - message: Init
-    reason: Init
-    status: Unknown
-    type: ConfigureCephClientReady
-  - message: Init
-    reason: Init
-    status: Unknown
-    type: ConfigureNetworkReady
   - message: Init
     reason: Init
     status: Unknown
diff --git a/tests/kuttl/tests/dataplane-deploy-no-nodes-test/00-assert.yaml b/tests/kuttl/tests/dataplane-deploy-no-nodes-test/00-assert.yaml
index 3190b6b..90a0ae0 100644
--- a/tests/kuttl/tests/dataplane-deploy-no-nodes-test/00-assert.yaml
+++ b/tests/kuttl/tests/dataplane-deploy-no-nodes-test/00-assert.yaml
@@ -53,14 +53,6 @@ status:
     severity: Info
     status: "False"
     type: Ready
-  - message: Init
-    reason: Init
-    status: Unknown
-    type: ConfigureCephClientReady
-  - message: Init
-    reason: Init
-    status: Unknown
-    type: ConfigureNetworkReady
   - message: Init
     reason: Init
     status: Unknown
diff --git a/tests/kuttl/tests/dataplane-deploy-no-nodes-test/01-assert.yaml b/tests/kuttl/tests/dataplane-deploy-no-nodes-test/01-assert.yaml
index bebd6a6..16fc3de 100644
--- a/tests/kuttl/tests/dataplane-deploy-no-nodes-test/01-assert.yaml
+++ b/tests/kuttl/tests/dataplane-deploy-no-nodes-test/01-assert.yaml
@@ -46,14 +46,6 @@ status:
     reason: Ready
     status: "True"
     type: Ready
-  - message: Init
-    reason: Init
-    status: Unknown
-    type: ConfigureCephClientReady
-  - message: Init
-    reason: Init
-    status: Unknown
-    type: ConfigureNetworkReady
   - message: ConfigureOS ready
     reason: Ready
     status: "True"
```

File: api/v1beta1/openstackdataplanenode_types.go [hunk 2]
```diff
@@ -102,17 +102,27 @@ func (instance *OpenStackDataPlaneNode) InitConditions() {
 	cl := condition.CreateList(
 		condition.UnknownCondition(condition.DeploymentReadyCondition, condition.InitReason, condition.InitReason),
 		condition.UnknownCondition(SetupReadyCondition, condition.InitReason, condition.InitReason),
-		condition.UnknownCondition(ConfigureNetworkReadyCondition, condition.InitReason, condition.InitReason),
```
Comment (slagle) [inline api/v1beta1/openstackdataplanenode_types.go:105]: can we fix the kuttl tests as a result of this change?

File: api/v1beta1/openstackdataplanenode_types.go [hunk 3]
```diff
@@ -102,17 +102,27 @@ func (instance *OpenStackDataPlaneNode) InitConditions() {
...
 		condition.UnknownCondition(ValidateNetworkReadyCondition, condition.InitReason, condition.InitReason),
 		condition.UnknownCondition(InstallOSReadyCondition, condition.InitReason, condition.InitReason),
 		condition.UnknownCondition(ConfigureOSReadyCondition, condition.InitReason, condition.InitReason),
 		condition.UnknownCondition(RunOSReadyCondition, condition.InitReason, condition.InitReason),
-		condition.UnknownCondition(ConfigureCephClientReadyCondition, condition.InitReason, condition.InitReason),
```
Comment (slagle) [inline api/v1beta1/openstackdataplanenode_types.go:110]: can we fix the kuttl tests as a result of this change?

**Review by slagle — CHANGES_REQUESTED** [review]

File: api/v1beta1/openstackdataplanenode_types.go (hunk 2 above)
Comment (rabi) [reply api/v1beta1/openstackdataplanenode_types.go:105]: Ok, Looks like https://github.com/openshift/release/pull/39222 merged only on last Friday.

**Review by slagle — COMMENTED** [review]
+2, thanks.

**Review by bshephar — APPROVED** [review]

**Pull request merged**